CommonMark/Parser/BlockMethods.cs

```diff
@@ -92,16 +93,355 @@ private static void BreakOutOfLists(ref Block blockRef, LineInfo line)
             {
                 while (container != null && container != b)
                 {
-                    Finalize(container, line);
+                    Finalize(container, line, settings);
                     container = container.Parent;
                 }
 
-                Finalize(b, line);
+                Finalize(b, line, settings);
                 blockRef = b.Parent;
             }
         }
 
-        public static void Finalize(Block b, LineInfo line)
+        static List<string> ParseTableLine(string line, StringBuilder sb)
+        {
+            var ret = new List<string>();
+
+            var i = 0;
+
+            if (i < line.Length && line[i] == '|') i++;
+
+            while (i < line.Length && char.IsWhiteSpace(line[i])) i++;
+
+            for (; i < line.Length; i++)
+            {
+                var c = line[i];
+                if (c == '\\')
+                {
+                    i++;
+                    continue;
+                }
+
+                if (c == '|')
+                {
+                    ret.Add(sb.ToString());
+                    sb.Clear();
+                }
+                else
+                {
+                    sb.Append(c);
```

Comment:
Instead of appending char by char, it should append whole substrings. See [InlineMethods](https://github.com/Knagis/CommonMark.NET/blob/master/CommonMark/Parser/InlineMethods.cs#L180) that contain multiple examples on how that could be done.